vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/AndroidDeviceTestRunner.kt

```diff
@@ -103,27 +97,38 @@ class AndroidDeviceTestRunner(private val device: AndroidDevice) {
         }
     }
 
-    private fun prepareTestRunner(configuration: Configuration,
-                                  androidConfiguration: AndroidConfiguration,
-                                  info: InstrumentationInfo,
-                                  testBatch: TestBatch): RemoteAndroidTestRunner {
-
-        val runner = RemoteAndroidTestRunner(info.instrumentationPackage, info.testRunnerClass, device.ddmsDevice)
-
+    private fun prepareTestRunner(
+        configuration: Configuration,
+        androidConfiguration: AndroidConfiguration,
+        info: InstrumentationInfo,
+        testBatch: TestBatch
+    ): RemoteAndroidTestRunner {
         val tests = testBatch.tests.map {
             "${it.pkg}.${it.clazz}#${it.method}"
         }.toTypedArray()
 
-        logger.debug { "tests = ${tests.toList()}" }
+        logger.debug { "device = [${device.serialNumber}]; tests = [${tests.toList()}]" }
 
+        // Single test timeout shouldn't be less then 'testOutputTimeoutMillis' and shouldn't be more than 'testBatchTimeoutMillis'
+        val testTimeout = min(
```

Comment:
This makes sense only if tests do not produce _any_ output at all until completion, is that really so?

Reply:
From what I've seen it is true at least for tests that are the problem in the current state of Marathon. But from what I've seen no log from the device is passed to Marathon until the test fails.